lib/fs.js

```diff
@@ -905,16 +905,47 @@ function symlink(target, path, type_, callback_) {
   validatePath(target, 'target');
   validatePath(path);
 
-  const flags = stringToSymlinkType(type);
   const req = new FSReqCallback();
   req.oncomplete = callback;
 
+  if (isWindows && type === null) {
+    let absoluteTarget;
+    try {
+      // Symlinks targets can be relative to the newly created path.
+      // Calculate absolute file name of the symlink target, and check
+      // if it is a directory. Ignore resolve error to keep symlink
+      // errors consistent between platforms if invalid path is
+      // provided.
+      absoluteTarget = pathModule.resolve(path, '..', target);
+    } catch (err) { } // eslint-disable-line no-unused-vars
```

Comment:
Any reason not to us  this instead?:

```suggestion
    } catch { } 
```

Then you don't need to disable the ESLint rule. The usual reason to avoid that `catch` syntax would be to simplify backporting to 8.x and 6.x  but this is a `semver-major` so that's not an issue here.